src/main/java/org/saltyrtc/client/signaling/Signaling.java

```diff
@@ -548,16 +552,25 @@ public synchronized void handleCallbackError(WebSocket websocket, Throwable caus
         // Reset timeout
         this.wsConnectTimeout = this.wsConnectTimeoutInitial;
 
+        // Create factory
+        // Note: We prefer the SSLSocketFactory over the SSLContext, if set.
+        //       They are mutually exclusive due to the way the WebSockets library uses them.
```

Comment:
It is documented: 

```
    /**
     * Return a non-null value if the WebSocket should use the SSLSocketFactory returned by this
     * method instead of the one associated to the SSLContext returned by getSSLContext.
     */
```

Reply:
But I'm now throwing in case both are set.